lib/streamlit/delta_generator.py

```diff
@@ -573,12 +573,36 @@ def _block(
         # Prevent nested columns & expanders by checking all parents.
         block_type = block_proto.WhichOneof("type")
         # Convert the generator to a list, so we can use it multiple times.
-        parent_block_types = frozenset(dg._parent_block_types)
-        if block_type == "column" and block_type in parent_block_types:
-            raise StreamlitAPIException(
-                "Columns may not be nested inside other columns."
-            )
-        if block_type == "expandable" and block_type in parent_block_types:
+        parent_block_types = list(dg._parent_block_types)
+        num_of_columns = len(
+            [
+                parent_block
+                for parent_block in parent_block_types
+                if "column" in str(parent_block)
+            ]
+        )
+        parent_block_types = frozenset(parent_block_types)
+        if block_type == "column":
+            if (
+                len(
+                    [
+                        parent_block
+                        for parent_block in parent_block_types
+                        if "1" in str(parent_block)
```

Comment:
Reading this, I don't know what "1" means in this context. Why a presence of "1" in the parent_block should change the logic?

Maybe you can extract lines 586-594 into a separate function with a meaningful name so the condition can be more readable?

Reply:
Right 👍 it was not readable, I've:
* moved it to `def _check_if_one_of_parents_is_a_sidebar`
* used `RootContainer.SIDEBAR` variable which is much more descriptive

Thank you 🙂